pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -180,8 +195,9 @@ func TestCreatePod(t *testing.T) {
 	}
 
 	for _, item := range testCases {
+		defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.PodDisruptionConditions, item.enablePodDisruptionConditions)()
```

Comment:
you have to run this as a subtest , or this defer only will be executed after the whole loop , not after each test case
https://go.dev/blog/subtests

Reply:
Adjusted